File: 203. Remove Linked List Elements/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode dummy{0};
        dummy.next = head;
        ListNode* curr = &dummy;
        while (curr) {
        ListNode* next = curr->next;
        while (next && next->val == val) 
            next = next->next;
        curr->next = next;
        curr = next;
        }
        return dummy.next;
    }
};

File: 203. Remove Linked List Elements/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null) 
            return null;
        ListNode curr = new ListNode(0), pre, now = head;
        curr.next = head; 
        pre = curr;
        while(now !=null){
            if(now.val == val) {
                pre.next = now.next; 
                now = now.next; 
            }
            else {
                pre = now; 
                now = now.next;}
        }
        return curr.next;
    }
}
